src/lib/helpers.ts

```diff
@@ -236,3 +236,44 @@ export function retryable<T>(
 
   return promise
 }
+
+function dec2hex(dec: number) {
+  return ('0' + dec.toString(16)).substr(-2)
+}
+
+// Functions below taken from: https://stackoverflow.com/questions/63309409/creating-a-code-verifier-and-challenge-for-pkce-auth-on-spotify-api-in-reactjs
+export async function generatePKCEVerifier() {
+  const verifierLength = 56
+  const array = new Uint32Array(verifierLength)
+  if (!isBrowser()) {
+    for (let i = 0; i < verifierLength; i++) {
+      array[i] = Math.floor(Math.random() * 256)
+    }
+  }
+  return Array.from(array, dec2hex).join('')
+}
+
+async function sha256(randomString: string) {
+  const encoder = new TextEncoder()
+  const encodedData = encoder.encode(randomString)
+  if (!isBrowser()) {
+    return sha256CryptoJS(randomString).toString()
+  }
+  const hash = await window.crypto.subtle.digest('SHA-256', encodedData)
+  let hashedString = ''
+  const bytes = new Uint8Array(hash)
+  const len = bytes.byteLength
+  for (let i = 0; i < len; i++) {
+    hashedString += String.fromCharCode(bytes[i])
+  }
+  return hashedString
+}
+
+function base64urlencode(str: string) {
+  return btoa(str).replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '')
+}
```

Comment:
I feel like this function is already in helpers somewhere.

Reply:
Do you recall where exactly? There's a `decodebase64URL`  and a regex but can't seem to find a function which encodes